recommender/recommender.py

```diff
@@ -443,36 +474,35 @@ def add_resource(self, data, _suffix=''):
         for field in self.resource_content_fields:
             result[field] = data[field]
 
+        resource_id = stem_url(data['url'])
         # check url for redundancy
-        for recommendation in self.recommendations:
-            if self.check_redundancy(recommendation['url'], data['url']):
-                result['error'] = ('The resource you are attempting to ' +
-                                   'provide has already existed')
-                for field in self.resource_content_fields:
-                    result['dup_' + field] = recommendation[field]
-                result['dup_id'] = recommendation['id']
-                result['Success'] = False
-                tracker.emit('add_resource', result)
-                return result
-            
+        if resource_id in self.recommendations:
+            result['error'] = ('The resource you are attempting to ' +
+                               'provide has already existed')
```

Comment:
XBlocks doesn't support i18n, or at least did not when we last spoke to Cale. There was an interrim prototype dummy service (which just returned the string). He suggested we omit i18n until it was more mature. Has this changed?